evalml/tests/automl_tests/test_automl.py

```diff
@@ -144,37 +144,37 @@ def test_pipeline_limits(mock_fit_binary, mock_score_binary,
     mock_score_multi.return_value = {'Log Loss Multiclass': 1.0}
     mock_score_regression.return_value = {'R2': 1.0}
 
-    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=automl_type, max_iterations=1)
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=automl_type, optimize_thresholds=False, max_iterations=1)
```

Comment:
For most of these tests, I choose to set `optimize_threhsholds=False` rather than patch `predict_proba`, `optimize_thresholds`, and `_encode_targets`.